org.eclipse.scanning.device.ui/src/org/eclipse/scanning/device/ui/composites/MultiStepComposite.java

```diff
@@ -0,0 +1,130 @@
+import org.eclipse.scanning.api.points.models.StepModel;
+import org.eclipse.scanning.api.scan.ScanningException;
+import org.eclipse.scanning.device.ui.util.SortNatural;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Label;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MultiStepComposite extends Composite {
+	
+	private static Logger logger = LoggerFactory.getLogger(MultiStepComposite.class);
+
+	// UI
+    private final ComboWrapper                  name;
+	private final VerticalListEditor<StepModel> steps;
+	private IScannableDeviceService scannableConnectorService;
+	private StepModelComposite stepComposite;
+
+	public MultiStepComposite(Composite parent, int style) throws ScanningException {
+		super(parent, style);
+		
+		setLayout(new GridLayout(2, false));
+		
+		Label label = new Label(this, SWT.NONE);
+		label.setText("Scannable");
+		
+		this.name = new ComboWrapper(this, SWT.READ_ONLY);
+		name.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, false));
+		name.addValueListener(new ValueAdapter("unitUpdater") {			
+			@Override
+			public void valueChangePerformed(ValueEvent e) {
+				updateUnits(e.getValue());
+			}
+		});
+			
+		// List of ExampleItems
+		this.steps = new VerticalListEditor<>(this, SWT.NONE);
+		steps.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, false, 2, 1));
+		steps.setMinItems(0);
+		steps.setMaxItems(10);
+		steps.setEditorClass(StepModel.class); // Must match generic!
+		steps.setBeanConfigurator((bean, previous, context)->contiguous(bean, previous));
+		steps.setListHeight(80);		
+		steps.setRequireSelectionPack(false);
+		steps.setTemplateName("Step");
+		
+		stepComposite = new StepModelComposite(this, SWT.NONE);
+		stepComposite.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true, 2, 1));
+		
+		steps.setEditorUI(stepComposite);
+
+	}
+	
+	protected boolean updateUnits(Object value) {
+		if (value==null) return false;
+		if (scannableConnectorService==null) return false;
+		try {
+			IScannable<Number> scannable = scannableConnectorService.getScannable(value.toString());
+			if (scannable!=null) {
+				stepComposite.setScannable(scannable);
+				return true;
+			}
+		} catch (Exception ne) {
+			logger.error("Cannot setup units!", ne);
+		}
+		return false;
+	}
+
+	@Inject
+	public void setScannableService(IScannableDeviceService service, MultiStepModel model) throws ScanningException {
+		
+		this.scannableConnectorService = service;
+		List<String> names = service.getScannableNames();
+		List<String> items = IFilterService.DEFAULT.filter("org.eclipse.scanning.scannableFilter", names);
+		Collections.sort(items, new SortNatural<>(false));
+		name.setItems(items.toArray(new String[items.size()]));
+		if (model.getName()!=null && items!=null) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2589)